src/main/java/seedu/findvisor/logic/commands/DeleteTagCommand.java

```diff
@@ -0,0 +1,146 @@
+
+    public static final String COMMAND_WORD = "deletetag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the tag associated with a particular person "
+            + "identified by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "[" + PREFIX_TAG + "TAG]\n"
+            + "Example: " + COMMAND_WORD + " 1 t/tag";
+
+    public static final String MESSAGE_DELETE_TAG_SUCCESS = "Deleted tag %1$s for Person: %2$s";
+    public static final String MESSAGE_CANNOT_FIND_TAG = "There is no tag %1$s for Person: %2$s";
+    public static final String MESSAGE_DELETE_PARTIAL_TAG = "Deleted tag %1$s for Person: %2$s\n"
+            + "There is no tag %3$s for Person: %2$s";
+
+    private final Index targetIndex;
+    private final Set<Tag> targetTags;
+
+    /**
+     * Creates an DeleteTagCommand to delete an existing tag with the person at the
+     * specified {@code Index}
+     */
+    public DeleteTagCommand(Index targetIndex, Set<Tag> targetTags) {
+        requireNonNull(targetIndex);
+        requireNonNull(targetTags);
+
+        this.targetIndex = targetIndex;
+        this.targetTags = targetTags;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+        Set<Tag> tagsOfPerson = personToEdit.getTags();
+        Set<Tag> tagsToDelete = new HashSet<>();
+        Set<Tag> tagsToReport = new HashSet<>();
+        for (Tag tag : targetTags) {
+            if (tagsOfPerson.contains(tag)) {
+                tagsToDelete.add(tag);
+            } else {
+                tagsToReport.add(tag);
+            }
+        }
+
+        if (tagsToReport.size() == 0) {
+            Person editedPerson = createEditedPerson(personToEdit, targetTags);
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(String.format(MESSAGE_DELETE_TAG_SUCCESS, targetTags, editedPerson.getName()));
+        } else if (tagsToDelete.size() == 0) {
+            return new CommandResult(String.format(MESSAGE_CANNOT_FIND_TAG, targetTags, personToEdit.getName()));
+        } else {
+            Person editedPerson = createEditedPerson(personToEdit, tagsToDelete);
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(String.format(MESSAGE_DELETE_PARTIAL_TAG, tagsToDelete,
+                    editedPerson.getName(), tagsToReport));
+        }
```

Comment:
Since in the case where `tagsToReport.size() == 0`, `tagsToDelete` will be the same as `tagsToReport`, I think this would be clearer and more concise
```suggestion
        if (tagsToDelete.size() == 0) {
            return new CommandResult(String.format(MESSAGE_CANNOT_FIND_TAG, targetTags, personToEdit.getName()));
        }

        Person editedPerson = createEditedPerson(personToEdit, tagsToDelete);
        model.setPerson(personToEdit, editedPerson);
        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);

        if (tagsToReport.size() == 0) {
            return new CommandResult(String.format(MESSAGE_DELETE_TAG_SUCCESS, targetTags, editedPerson.getName()));
        }
        return new CommandResult(String.format(MESSAGE_DELETE_PARTIAL_TAG, tagsToDelete,
                editedPerson.getName(), tagsToReport));
```